### PR TITLE
Add eslint infrastructure

### PR DESCRIPTION
**Pull request by Amir-61**
* Add eslint infrastructure

Connect to https://github.com/strongloop-internal/scrum-loopback/issues/1120

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "extends": "loopback"
+}
```
Comment (pthieu) [inline .eslintrc]: needs an empty line after this.

**Review by pthieu — CHANGES_REQUESTED** [review]

**Comment by pthieu** [comment]
Did you try an `npm test`? Looks like `eslint` isn't triggering.

**Comment by Amir-61** [comment]
> Did you try an npm test? Looks like eslint isn't triggering.

Yes I tried and it does trigger eslint and solved the failures manually

```
  mixin example
    ✓ should load the module mixin
    ✓ should load the local mixin
    ✓ should be assigned `squirrel` before saving


  3 passing (11ms)


> loopback-example-mixins@1.0.0 posttest /Users/jafaria/workspace/loopback-example-mixins
> eslint .
```

File: package.json [hunk 2]
```diff
@@ -6,7 +6,7 @@
   "scripts": {
     "start": "slc run",
     "test": "mocha -R spec test",
-    "pretest": "jshint ."
+    "posttest": "eslint ."
```
Comment (pthieu) [inline package.json:9]: should be `pretest`.
no point running tests if lint fails.

**Review by pthieu — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (pthieu) [reply package.json:9]: Talked to @superkhau: IBM has chosen `posttest` for the consideration of the external contributions.

Comment (Amir-61) [reply package.json:9]: Actually for all Loopback repos we use `posttest`:

> Modify package.json and/or Gruntfile to ensure that linter (eslint) is run after the tests pass. This typically means adding a "posttest": "eslint" script. If the original package contains a pretest script to run the linter, then remove the pretest script in favor of posttest.

Source: https://github.com/strongloop/loopback.io/blob/gh-pages/pages/en/contrib/eslint-guide.md

**Pull request merged**